frontend/src/app/testResults/DownloadResultsCsvModal.tsx

```diff
@@ -0,0 +1,162 @@
+    filterParams.patientId ||
+    filterParams.startDate ||
+    filterParams.endDate ||
+    filterParams.role ||
+    filterParams.result ||
+    (filterParams.filterFacilityId &&
+      filterParams.filterFacilityId !== activeFacilityId);
+
+  const variables: ResultsQueryVariables = {
+    facilityId:
+      filterParams.filterFacilityId === ALL_FACILITIES_ID
+        ? null
+        : filterParams.filterFacilityId || activeFacilityId,
+    pageNumber: 0,
+    pageSize: totalEntries,
+    ...filterParams,
+  };
+
+  const getResults = useImperativeQuery(GetFacilityResultsForCsvDocument, {
+    fetchPolicy: "no-cache",
+  });
+
+  const downloadResults = async () => {
+    setLoading(true);
+    const { data, error } = await getResults(variables);
+    if (error) {
+      showError("Error downloading results", error.message);
+      setLoading(false);
+    } else {
+      const results = parseDataForCSV(data.testResults, multiplexEnabled);
+      setResults(results);
+      setLoading(false);
+      csvLink?.current?.link.click();
+      closeModal();
+    }
+  };
+
+  return (
+    <Modal
+      isOpen={modalIsOpen}
+      style={{
+        content: {
+          maxHeight: "90vh",
+          width: "40em",
+          position: "initial",
+        },
+      }}
+      overlayClassName="prime-modal-overlay display-flex flex-align-center flex-justify-center"
+      contentLabel="Download test results"
+      ariaHideApp={process.env.NODE_ENV !== "test"}
+    >
+      <div className="border-0 card-container">
+        <div className="display-flex flex-justify">
+          <h1 className="font-heading-lg margin-top-05 margin-bottom-0">
+            {disableDownload
+              ? "Too many results selected"
+              : "Download test results"}
+          </h1>
+          <button
+            onClick={closeModal}
+            className="close-button"
+            aria-label="Close"
+          >
+            <span className="fa-layers">
+              <FontAwesomeIcon icon={"circle"} size="2x" inverse />
+              <FontAwesomeIcon icon={"times-circle"} size="2x" />
+            </span>
+          </button>
+        </div>
+        <div className="border-top border-base-lighter margin-x-neg-205 margin-top-205"></div>
+        {disableDownload ? (
+          <div className="grid-row grid-gap">
+            <p>
+              Please filter test results and download again with 20,000 results
+              or fewer.
+            </p>
+          </div>
+        ) : (
+          <>
+            <div className="grid-row grid-gap">
+              {filtersPresent ? (
+                <p>Download results with current search filters applied?</p>
+              ) : (
+                <p>Download results without any search filters applied?</p>
+              )}
+            </div>
+            <div className="grid-row grid-gap">
+              <p id="sr-download-results-csv-text">
+                The CSV file will include {totalEntries} row
```

Comment:
⚠️ Added this since I noticed the grammar was off -- let me know if there is a better way to do this :)